internal/cmdparser/interface.go

```diff
@@ -42,4 +42,8 @@ type Command interface {
 	// the Command object to have access to the dependencies it needs, without
 	// having to manage them directly.
 	SetCrossCuttingConcerns(dependency.Options)
+
+	// LogTelemtry is used to track the useful telemetry for the command
+	// To be enforced after the review
+	// LogTelemtry()
```

Comment:
>// LogTelemtry() [](http://example.com/codeflow?start=1&length=16)

if we switch to a new method, it should probably pass the map as a parameter and not require the command to invoke telemetry methods directly.